src/Controller.php

```diff
@@ -33,11 +34,12 @@ public function __construct(
 
     protected function render(string $view, array $parameters = []): ResponseInterface
     {
-        $response = $this->responseFactory->createResponse();
-        $content = $this->view->render($view, $parameters, $this);
-        $response->getBody()->write($this->renderContent($content));
+        $controller = $this;
+        $contentRenderer = function () use ($view, $parameters, $controller) {
+            return $controller->renderContent($controller->view->render($view, $parameters, $controller));
+        };
 
-        return $response;
+        return $this->factory->create(DeferredResponse::class, [$contentRenderer]);
```

Comment:
Likely could be made more explicit about dependencies:

1. Create `DefferedResponseFactory` that creates `DeferredResponse`.
2. Inject it into constructor.

Reply:
Done